Core/clim-basic/extended-streams/recording.lisp

```diff
@@ -1150,42 +1151,50 @@ were added."
 ;;; which calls the medium drawing function based on the recorded
 ;;; slots.
 
-(defmacro def-grecording (name ((&rest mixins) &rest args)
-                               (&key (class t)
-                                     (medium-fn t)
-                                     (replay-fn t)) &body body)
-  (let ((method-name (symbol-concat '#:medium- name '*))
-        (class-name (symbol-concat name '#:-output-record))
-        (medium (gensym "MEDIUM"))
-        (class-vars `((stream :initarg :stream)
-                      ,@(loop for arg in args
-                           collect `(,arg
-                                     :initarg ,(intern (symbol-name arg)
-                                                       :keyword))))))
-    `(progn
-       ,@(when class
-           `((defclass ,class-name (,@mixins standard-graphics-displayed-output-record)
-               ,class-vars)
-             (defmethod initialize-instance :after ((graphic ,class-name) &key)
-               (with-slots (stream ink clipping-region line-style text-style ,@args)
-                   graphic
-                 (let ((medium (sheet-medium stream)))
-                   (setf (rectangle-edges* graphic) (progn ,@body)))))))
-       ,@(when medium-fn
-           `((defmethod ,method-name :around ((stream output-recording-stream) ,@args)
-                        ;; XXX STANDARD-OUTPUT-RECORDING-STREAM ^?
-                        (generate-medium-recording-body ,class-name ,args))))
-       ,@(when replay-fn
-           `((defmethod replay-output-record ((record ,class-name) stream
-                                              &optional (region +everywhere+)
-                                                (x-offset 0) (y-offset 0))
-               (declare (ignore x-offset y-offset region))
-               (with-slots (,@args) record
-                 (let ((,medium (sheet-medium stream)))
-                   ;; Graphics state is set up in :around method.
-                   (,method-name ,medium ,@args)))))))))
-
-(def-grecording draw-point ((gs-line-style-mixin) point-x point-y) ()
+(defmacro def-grecording (name-and-options (&rest mixins) (&rest args)
+                          &body body)
```

Comment:
I like this change of the macro lambda list, however it will require a longer comment explaining things.